quodlibet/quodlibet/qltk/lyrics.py

```diff
@@ -113,23 +113,45 @@ def __search(self, song, buffer, refresh, add):
             GLib.idle_add(buffer.set_text, text)
             GLib.idle_add(refresh.set_sensitive, True)
 
-    def __save(self, save, lyricname, buffer, delete):
+    def __save(self, save, song, buffer, delete):
+        start, end = buffer.get_bounds()
+        text = buffer.get_text(start, end, True)
+
+        # First, write back to the tags.
+        song["lyrics"] = text
+        try:
+            song.write()
+        except NotImplementedError:
+            pass
+
+        # Then, write to file.
```

Comment:
Hm? We write the modified text to file's tags, then write the same text to the external file (previous behavior).

BTW, it would be good if you told me how to detect whether the format handler really writes lyrics to tags (and does not ignore them, as it was with ID3 before this patch). If it really does, then we could skip writing to the file altogether. Or keep it that way for forward-compatibility with earlier Quod Libet versions?